platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/MarkerViewManager.java

```diff
@@ -503,6 +503,21 @@ public void invalidateViewMarkersInVisibleRegion() {
                 }
               }
 
+              adaptedView.setOnClickListener(new View.OnClickListener() {
```

Comment:
Just a thought, we should refactor `invalidateViewMarkersInVisibleRegion` method. The longer a procedure is, the more difficult it’s to understand. It's an easy refactor, 99% of the times extract a method. What do you think?